template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Index.cshtml.cs

```diff
@@ -113,15 +116,17 @@ public class InputModel
                 return Page();
             }
 
-            var user = await _userManager.GetUserAsync(HttpContext.User);
+            var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
                 return RedirectToPage("/Error");
             }
 
             var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
-            var callbackUrl = Url.EmailConfirmationLink(user.Id, code, Request.Scheme);
-            await _emailSender.SendEmailConfirmationAsync(Input.Email, callbackUrl);
+            var userId = await _userManager.GetUserIdAsync(user);
```

Comment:
@HaoK Would it be better to use the user manager to pull userId and username out of the principal? I.e. `_userManager.GetUserName(User)` and `_userManager.GetUserId(User)`